### PR TITLE
TOFTOF GUI: improve save reduced data options

### LINKED ISSUE #22712: TOFTOF GUI: save reduced data
Refactor `save reduced data` block

- `SaveNXSPE` should be available only for `S(2theta,W)`  
- `SaveAscii` should be active

The block should look approximately like:

Directory __________________ (Browse)

[ ] Save S(Q, W):           [ ] NeXus    [ ] Ascii

[ ] Save S(2theta, W):   [ ] NXSPE    [ ] NeXus   [ ] Ascii 

For `SaveAscii` option
- `SaveAscii(..., SpectrumMetaData='Q')` should be used to save the `S(Q, W)`  
- `SaveAscii(..., SpectrumMetaData='Angle')` should be used to save the `S(2theta, W)`  

labels: QENS, GUI

@JoachimCoenen will work on it

### PR DESCRIPTION
**Pull request by JoachimCoenen**
**Description of work.**

Refactored save reduced data block:
 - data formats can be chosen for `S(Q, W)` and `S(2theta, W)` independently
 - `ascii` option has been activated

**To test:**

Start the TOFTOF data reduction GUI: `Interfaces`&rarr;`Direct`&rarr;`DGS Reduction`. Choose facility MLZ and instrument TOFTOF.

Get the test data
[testdata.zip](https://github.com/mantidproject/mantid/files/1356265/testdata.zip)

and make sure, that those workspaces and file formats are saved, whos corresponding checkboxes are checked *and enabled*.

Fixes #22712. 

Does this update require release notes?
- [x] Yes
- [ ] No

---
#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####

- [ ] Is the code of an acceptable quality?
- [ ] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)?
- [ ] Are the unit tests small and test the class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- [ ] Do changes function as described? Add comments below that describe the tests performed?
- [ ] Do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Comment by reimundILL** [comment]
Please correct me if the milestone is not correct. I added the `In Progress` flag since documentation updates are still missing?

**Comment by mganeva** [comment]
@JoachimCoenen please, update the release notes:

```
SaveAscii option is now available in the TOFTOF data reduction GUI.
```

@reimundILL For the moment, we don't have a GUI documentation at all. It will be a separate issue.

**Comment by mganeva** [comment]
@reimundILL I've tested the PR. It is good for me. Could you please remove `In progress` label after the release notes will get updated?

File: docs/source/release/v3.14.0/index.rst [hunk 1]
```diff
@@ -0,0 +1,51 @@
+.. _v3.13.0:
```
Comment (reimundILL) [inline docs/source/release/v3.14.0/index.rst]: v3.13 in this file?

**Comment by reimundILL** [comment]
Maybe it would be good to only reference the open issue for the GUI documentation here?

**Comment by mganeva** [comment]
@reimundILL I can create an issue and refer to it here, but since there is no instrument scientist request for the GUI documentation, it will be quite a low priority issue. The TOFTOF GUI is quite simple with pop-up 'tips'.

File: scripts/Interface/reduction_gui/widgets/toftof/toftof_setup.py [hunk 2]
```diff
@@ -317,18 +338,19 @@ def _onSaveDir(self):
...
 
-    def _onBinEon(self, onVal):
-        if not onVal:
-            self.chkNxspe.setChecked(False)
-            self.chkReplaceNaNs.setChecked(False)
-            self.binQon.setChecked(False)
-        for widget in (self.binEstart, self.binEstep, self.binEend, self.chkCreateDiff, self.chkNxspe, self.binQon,
-                       self.binQstart, self.binQstep, self.binQend, self.chkReplaceNaNs, self.chkSofQW):
-            widget.setEnabled(onVal)
+    #def _onBinEon(self, onVal):
```
Comment (reimundILL) [inline scripts/Interface/reduction_gui/widgets/toftof/toftof_setup.py]: There is some commented code here that can be deleted? It is available via git.

**Review by vreimund — APPROVED** [review]
Save options added to TOFTOF GUI and tested successfully by @mganeva, There are no related tests and a documentation may or may not be future perspective. Eventually, commented code can be deleted.

**Comment by mganeva** [comment]
@JoachimCoenen Please, delete the commented code.
@reimundILL Thank you for the review. The documentation is not planned for the moment, but the PR for tests is here: #22836.

**Pull request merged**